From: Sergey Senozhatsky <senozhatsky@chromium.org>
Subject: zram: clear trailing bytes of compressed writeback pages
Date: Mon, 25 May 2026 13:46:02 +0900

When compressed writeback is available writtenback pages contain garbage
PAGE_SIZE - obj_size trailing bytes.  Zero out those trailing bytes before
writeback to a backing device.

Link: https://lore.kernel.org/20260525044639.1888561-3-senozhatsky@chromium.org
Signed-off-by: Sergey Senozhatsky <senozhatsky@chromium.org>
Suggested-by: Brian Geffon <bgeffon@google.com>
Cc: Jens Axboe <axboe@kernel.dk>
Cc: Minchan Kim <minchan@kernel.org>
Cc: Richard Chang <richardycc@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 drivers/block/zram/zram_drv.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/block/zram/zram_drv.c~zram-clear-trailing-bytes-of-compressed-writeback-pages
+++ a/drivers/block/zram/zram_drv.c
@@ -2134,6 +2134,8 @@ static int read_from_zspool_raw(struct z
 	zs_obj_read_end(zram->mem_pool, handle, size, src);
 	zcomp_stream_put(zstrm);
 
+	memzero_page(page, size, PAGE_SIZE - size);
+
 	return 0;
 }
 #endif
_
